test/system/src/test/java/io/pravega/test/system/MultiReaderWriterTest.java

```diff
@@ -0,0 +1,267 @@
+import io.pravega.client.stream.ScalingPolicy;
+import io.pravega.client.stream.StreamConfiguration;
+import io.pravega.client.stream.impl.Controller;
+import io.pravega.client.stream.impl.JavaSerializer;
+import io.pravega.test.system.framework.Environment;
+import io.pravega.test.system.framework.SystemTestRunner;
+import io.pravega.test.system.framework.services.BookkeeperService;
+import io.pravega.test.system.framework.services.PravegaControllerService;
+import io.pravega.test.system.framework.services.PravegaSegmentStoreService;
+import io.pravega.test.system.framework.services.Service;
+import io.pravega.test.system.framework.services.ZookeeperService;
+import mesosphere.marathon.client.utils.MarathonException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+import lombok.Cleanup;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+@Slf4j
+@RunWith(SystemTestRunner.class)
+public class MultiReaderWriterTest extends AbstractScaleTests {
+
+    private static final String STREAM_NAME = "testMultiReaderWriterStream";
+    private static final int NUM_WRITERS = 20;
+    private static final int NUM_READERS = 20;
+    private static final long NUM_EVENTS = 20000;
+    private AtomicBoolean stopReadFlag;
+    private AtomicLong eventData;
+    private AtomicLong eventReadCount;
+    private ConcurrentLinkedQueue<Long> eventsReadFromPravega;
+    private Service controllerInstance = null;
+    private Service segmentStoreInstance = null;
+
+    @Environment
+    public static void initialize() throws InterruptedException, MarathonException, URISyntaxException {
+
+        //1. Start 1 instance of zookeeper
+        Service zkService = new ZookeeperService("zookeeper");
+        if (!zkService.isRunning()) {
+            zkService.start(true);
+        }
+        List<URI> zkUris = zkService.getServiceDetails();
+        log.debug("Zookeeper service details: {}", zkUris);
+        //get the zk ip details and pass it to bk, host, controller
+        URI zkUri = zkUris.get(0);
+
+        //2. Start 3 bookies
+        Service bkService = new BookkeeperService("bookkeeper", zkUri);
+        if (!bkService.isRunning()) {
+            bkService.start(true);
+        }
+        List<URI> bkUris = bkService.getServiceDetails();
+        log.debug("Bookkeeper service details: {}", bkUris);
+
+        //3. start 2 instances of pravega controller
+        Service conService = new PravegaControllerService("controller", zkUri, 2, 0.1, 700.0);
```

Comment:
No need to hard code the CPU and memory, we can use the `scaleService` api to scale it to two instances.